common/src/main/java/com/sequenceiq/cloudbreak/util/URLUtils.java

```diff
@@ -16,4 +18,9 @@ public static String readUrl(String url) throws IOException {
         }
         return IOUtils.toString(new URL(url));
     }
+
+    public static String encodeString(String text) {
+        return URLEncoder.encode(text, StandardCharsets.UTF_8).replaceAll("\\+", "%20");
```

Comment:
I don't really get this. `+` would be converted to `%2B` by `URLEncoder`. Also I would expect this class would encode it to a safe URL and we don't have to do any magic afterward.

Reply:
This one is tricky. From the class Javadoc of `URLEncoder`:

> The space character "   " is converted into a plus sign "+". 

Since the main purpose of that class is to produce {{application/x-www-form-urlencoded}} compliant content. The URL itself differs from this in that the spaces have to be percent-encoded. Hence the post-encoding replacement of the `+` with `%20`.

I actually ~stole~ borrowed this behavior from the CM Swagger Java client. Quoting the interesting part from `com.cloudera.api.swagger.HostsResourceApi.updateHostCall` (`cloudera-manager-api-swagger-7.4.1.jar`):
```
    public Call updateHostCall(String hostId, ApiHost body, final ProgressListener progressListener, ProgressRequestListener progressRequestListener) throws ApiException {
        String localVarPath = "/hosts/{hostId}".replaceAll("\\{hostId\\}", this.apiClient.escapeString(hostId.toString()));
...
```
where `com.cloudera.api.swagger.client.ApiClient.escapeString()` looks like:
```
    public String escapeString(String str) {
        try {
            return URLEncoder.encode(str, "utf8").replaceAll("\\+", "%20");
        } catch (UnsupportedEncodingException var3) {
            return str;
        }
    }
```

I used the overloaded method variant that accepts a `Charset` and does not throw `UnsupportedEncodingException`.